MINOR: Align assertFutureThrows method signature with JUnit conventions

Parkerhiphop opened the pull request:
Align assertFutureThrows method signature with JUnit conventions based on the suggestion from https://github.com/apache/kafka/pull/18296#discussion_r1900565069.

From `assertFutureThrows(Future<?> future, Class expectedCauseClassApiException)` to `assertFutureThrows(Class expectedCauseClassApiException, Future<?> future)`

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

Parkerhiphop committed:
MINOR: Align assertFutureThrows method signature with JUnit conventions

AndrewJSchofield commented on clients/src/test/java/org/apache/kafka/test/TestUtils.java:
```diff
@@ -583,12 +583,12 @@ public static Set<TopicPartition> generateRandomTopicPartitions(int numTopic, in
+    public static <T extends Throwable> T assertFutureThrows(Class<T> exceptionCauseClass, Future<?> future) {
         ExecutionException exception = assertThrows(ExecutionException.class, future::get);
         Throwable cause = exception.getCause();
         assertEquals(exceptionCauseClass, cause.getClass(),
```
This could probably be `assertInstanceOf` since we're cleaning up the use of assertions here.

Parkerhiphop replied:
Thanks for reviewing.
I've changed it to `assertInstanceOf` and merged the latest changes to resolve conflicts.

Parkerhiphop replied:
@AndrewJSchofield I just found [this discussion](https://github.com/apache/kafka/pull/18296#discussion_r1899687406) suggesting the adoption of `assertEquals` instead of `assertInstanceOf`. 

Is this still valid in the context of the cleanup process?

chia7712 replied:
Yes, in the previous PR we preferred to do strict check for the type of exception. Maybe we can keep the origin behavior but this PR can add comments to avoid potential confusion in the future.

AndrewJSchofield replied:
Makes sense to me.

Parkerhiphop replied:
Thanks @chia7712 for the advice.
I’ve reverted the assertEquals changes and added a comment above it.

AndrewJSchofield reviewed (changes requested):
Thanks for the PR. Just one change requested. Also, please merge the latest changes so we get a clean build with no conflicts. Thanks.

Parkerhiphop committed:
Replace assertEquals with assertInstanceOf

Parkerhiphop committed:
fix: format

Parkerhiphop committed:
Fix the missed signature changed

Parkerhiphop committed:
Revert to assertEquals and comment

Parkerhiphop committed:
Fix new test using assertFutureThrows

AndrewJSchofield commented:
@Parkerhiphop Please can you investigate why the tests timed out.

Parkerhiphop committed:
Merge branch 'upstream-trunk' into minor-align-assertfuturethrows-signature

Parkerhiphop committed:
Fix the missed signature changed

Parkerhiphop commented:
> @Parkerhiphop Please can you investigate why the tests timed out.

@AndrewJSchofield Thanks for reminding. I resolve it by merging the latest changes .

AndrewJSchofield commented on core/src/test/scala/integration/kafka/api/PlaintextAdminIntegrationTest.scala:
```diff
@@ -2225,14 +2227,14 @@ class PlaintextAdminIntegrationTest extends BaseAdminIntegrationTest {
-        assertTrue(describeWithFakeGroupResult.describedGroups().containsKey(fakeGroupId))
-        assertFutureThrows(describeWithFakeGroupResult.describedGroups().get(fakeGroupId),
-          classOf[GroupIdNotFoundException], s"Group $fakeGroupId not found.")
+          // Test that the fake group throws GroupIdNotFoundException
```
The indentation is changed in several changes in this file.

AndrewJSchofield commented on clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:
```diff
@@ -1493,7 +1493,7 @@ public void testInvalidTopicNames() throws Exception {
             Map<String, KafkaFuture<Void>> createFutures = env.adminClient().createTopics(newTopics).values();
             for (String sillyTopicName : sillyTopicNames) {
-                TestUtils.assertFutureThrows(createFutures .get(sillyTopicName), InvalidTopicException.class);
+                TestUtils.assertFutureThrows(InvalidTopicException.class, createFutures .get(sillyTopicName));
```
nit: Unnecessary space in `createFutures .get`.

AndrewJSchofield commented on clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:
```diff
@@ -1125,10 +1125,10 @@ public void testCreateTopicsRetryThrottlingExceptionWhenEnabledUntilRequestTimeO
-            ThrottlingQuotaExceededException e = TestUtils.assertFutureThrows(result.values().get("topic2"),
-                ThrottlingQuotaExceededException.class);
+            ThrottlingQuotaExceededException e = TestUtils.assertFutureThrows(ThrottlingQuotaExceededException.class, result.values().get("topic2")
+            );
```
nit: Orphaned `)` on its own line.

AndrewJSchofield commented on clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:
```diff
@@ -1152,10 +1152,10 @@ public void testCreateTopicsDontRetryThrottlingExceptionWhenDisabled() throws Ex
-            ThrottlingQuotaExceededException e = TestUtils.assertFutureThrows(result.values().get("topic2"),
-                ThrottlingQuotaExceededException.class);
+            ThrottlingQuotaExceededException e = TestUtils.assertFutureThrows(ThrottlingQuotaExceededException.class, result.values().get("topic2")
+            );
```
nit: Orphaned `)` on its own line.

AndrewJSchofield commented on clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:
```diff
@@ -1358,10 +1358,10 @@ public void testDeleteTopicsRetryThrottlingExceptionWhenEnabledUntilRequestTimeO
-            ThrottlingQuotaExceededException e = TestUtils.assertFutureThrows(result.topicNameValues().get("topic2"),
-                ThrottlingQuotaExceededException.class);
+            ThrottlingQuotaExceededException e = TestUtils.assertFutureThrows(ThrottlingQuotaExceededException.class, result.topicNameValues().get("topic2")
+            );
```
nit: Orphaned `)` on its own line.

AndrewJSchofield commented on clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:
```diff
@@ -1395,10 +1395,10 @@ public void testDeleteTopicsRetryThrottlingExceptionWhenEnabledUntilRequestTimeO
-            e = TestUtils.assertFutureThrows(resultIds.topicIdValues().get(topicId2),
-                    ThrottlingQuotaExceededException.class);
+            e = TestUtils.assertFutureThrows(ThrottlingQuotaExceededException.class, resultIds.topicIdValues().get(topicId2)
+            );
```
nit: Orphaned `)` on its own line.

AndrewJSchofield commented on clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:
```diff
@@ -1419,10 +1419,10 @@ public void testDeleteTopicsDontRetryThrottlingExceptionWhenDisabled() throws Ex
-            ThrottlingQuotaExceededException e = TestUtils.assertFutureThrows(result.topicNameValues().get("topic2"),
-                ThrottlingQuotaExceededException.class);
+            ThrottlingQuotaExceededException e = TestUtils.assertFutureThrows(ThrottlingQuotaExceededException.class, result.topicNameValues().get("topic2")
+            );
```
nit: Orphaned `)` on its own line.

AndrewJSchofield commented on clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:
```diff
@@ -1440,10 +1440,10 @@ public void testDeleteTopicsDontRetryThrottlingExceptionWhenDisabled() throws Ex
-            e = TestUtils.assertFutureThrows(resultIds.topicIdValues().get(topicId2),
-                    ThrottlingQuotaExceededException.class);
+            e = TestUtils.assertFutureThrows(ThrottlingQuotaExceededException.class, resultIds.topicIdValues().get(topicId2)
+            );
```
nit: Orphaned `)` on its own line.

AndrewJSchofield commented on clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:
```diff
@@ -1888,11 +1888,11 @@ public void testDescribeAcls() throws Exception {
-            TestUtils.assertFutureThrows(env.adminClient().describeAcls(UNKNOWN_FILTER).values(),
-                InvalidRequestException.class);
+            TestUtils.assertFutureThrows(InvalidRequestException.class, env.adminClient().describeAcls(UNKNOWN_FILTER).values()
+            );
```
nit: Orphaned `)` on its own line.

AndrewJSchofield commented on clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:
```diff
@@ -2625,10 +2625,10 @@ public void testCreatePartitionsRetryThrottlingExceptionWhenEnabledUntilRequestT
-            ThrottlingQuotaExceededException e = TestUtils.assertFutureThrows(result.values().get("topic2"),
-                ThrottlingQuotaExceededException.class);
+            ThrottlingQuotaExceededException e = TestUtils.assertFutureThrows(ThrottlingQuotaExceededException.class, result.values().get("topic2")
+            );
```
nit: Orphaned `)` on its own line.

AndrewJSchofield commented on clients/src/test/java/org/apache/kafka/clients/admin/KafkaAdminClientTest.java:
```diff
@@ -2653,10 +2653,10 @@ public void testCreatePartitionsDontRetryThrottlingExceptionWhenDisabled() throw
-            ThrottlingQuotaExceededException e = TestUtils.assertFutureThrows(result.values().get("topic2"),
-                ThrottlingQuotaExceededException.class);
+            ThrottlingQuotaExceededException e = TestUtils.assertFutureThrows(ThrottlingQuotaExceededException.class, result.values().get("topic2")
+            );
```
nit: Orphaned `)` on its own line.

AndrewJSchofield reviewed (changes requested):
Thanks for the update. A few more comments to address.

I know this is a slightly larger change, but there are about 40 instances of `assertFutureThrows` using the pattern `Errors.XXX_XXX.exception().getClass()` which should be entirely equivalent to `XxxXXxException.class`. For example, if XXX is `INVALID_REPLICA_ASSIGNMENT`, it would be `InvalidReplicaAssignmentException.class`. Do you think it would be good to changes these over to the class names for brevity? Just a suggestion which you can ignore.

Parkerhiphop committed:
Fix orphaned parentheses and unnecessary space

Parkerhiphop committed:
Direct reference exception for assertFutureThrows

Parkerhiphop committed:
Merge branch 'upstream-trunk' into minor-align-assertfuturethrows-signature

Parkerhiphop commented:
@AndrewJSchofield The suggestion is great!
Using the direct reference to the exception (`XxxXXxException.class`) makes the code more concise and readable.
Since I’ve already modified these lines to change the signature, I think this is a good time to make this update as well.
I’ve updated the code accordingly(also fix the lint issues above). 
Thanks for reviewing.

Parkerhiphop committed:
Remove unintened indentation

Parkerhiphop committed:
Remove unintened indentation

Parkerhiphop committed:
Merge branch 'upstream-trunk' into minor-align-assertfuturethrows-signature

AndrewJSchofield reviewed (approved):
Thanks for the PR. Looks good to me.
